lasagne/layers/merge.py

```diff
@@ -303,8 +303,9 @@ class ElemwiseMergeLayer(MergeLayer):
     Parameters
     ----------
     incomings : a list of :class:`Layer` instances or tuples
-        the layers feeding into this layer, or expected input shapes,
-        with all incoming shapes being equal
+        the layers feeding into this layer, or expected input shapes.
+        All input shapes must be equal or broadcastable. A dimension is
+        broadcastable if all non None dimensions are the same size or 1.
```

Comment:
> Should that give an error or just return None in the last dimensions?

No, it should return (32, 5) for the output shape. That's the only possible outcome. We can't raise an error because the run-time shape of input3 could be correct.